monitor/influxdb/init.sls

```diff
@@ -0,0 +1,45 @@
+
+#
+# Install and configure influxDB
+#
+
+
+influxdb_pkg:
+  pkg.installed:
+    - sources:
+      - influxdb: http://influxdb.s3.amazonaws.com/influxdb_0.9.2_amd64.deb
+
+/etc/opt/influxdb/influxdb.conf:
+  file:
+    - managed
+    - source: salt://monitor/etc/influxdb/influxdb.conf
+    - template: jinja
+    - makedirs: true
+    - require:
+      - pkg: influxdb_pkg
+
+{{ storage_dir }}/data:
```

Comment:
does this work?  I think you'd need something like `{{pillar.monitoring.storage_dir}}`, or `{% set storage_dir= salt['pillar.get']('monitoring:influxdb:storage_dir', '/mnt/influxdb') %}` at the top of the file.